### PR TITLE
Chores/typescript support

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
Add typescript support

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const express = require("express")
```
Comment (olivermrbl) [inline index.js]: todo: Remove

File: medusa-config.js [hunk 2]
```diff
@@ -44,8 +44,31 @@ const plugins = [
...
+    options: {
+      redisUrl: REDIS_URL
+    }
+  },
+  /*inventoryService: {
+    resolve: '@medusajs/inventory'
+  } ,
+  stockLocationService: {
+    resolve: '@medusajs/stock-location'
+  },*/
```
Comment (olivermrbl) [inline medusa-config.js]: todo: remove

File: medusa-config.js [hunk 3]
```diff
@@ -44,8 +44,31 @@ const plugins = [
...
+    options: {
+      redisUrl: REDIS_URL
+    }
+  },
+  cacheService: {
+    resolve: "@medusajs/cache-redis",
+    options: {
+      redisUrl: REDIS_URL
+    }
+  },
```
Comment (olivermrbl) [inline medusa-config.js]: todo: comment out

File: package.json [hunk 4]
```diff
@@ -4,36 +4,84 @@
...
+    "start:custom": "npm run build && node index.js",
     "build:admin": "medusa-admin build"
   },
   "dependencies": {
+    "@babel/preset-typescript": "^7.21.4",
     "@medusajs/admin": "rc",
-    "@medusajs/cache-inmemory": "rc",
-    "@medusajs/event-bus-local": "rc",
+    "@medusajs/cache-redis": "rc",
+    "@medusajs/event-bus-redis": "rc",
```
Comment (olivermrbl) [inline package.json]: todo: add default ones as well

File: package.json [hunk 5]
```diff
@@ -4,36 +4,84 @@
...
   "scripts": {
+    "clean": "./node_modules/.bin/rimraf dist",
+    "build": "npm run clean && tsc -p tsconfig.json",
+    "watch": "tsc --watch",
+    "test": "jest",
     "seed": "medusa seed -f ./data/seed.json",
-    "build": "babel src -d dist --extensions \".ts,.js\"",
-    "start": "medusa develop",
+    "start": "npm run build && medusa start",
+    "start:custom": "npm run build && node index.js",
```
Comment (olivermrbl) [inline package.json]: todo: remove

File: src/api/routes/store/index.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
+import * as cors from "cors"
```
Comment (olivermrbl) [inline src/api/routes/store/index.ts:1]: todo: One example is enough

File: src/models/custom.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { generateEntityId } from "@medusajs/utils"
```
Comment (olivermrbl) [inline src/models/custom.ts]: todo: Add in README in `src/models/README.md`

File: src/migrations/1617703530229-my-migration.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
```
Comment (olivermrbl) [inline src/migrations/1617703530229-my-migration.ts]: todo: Add in README in `src/migrations/README.md`

File: src/loaders/loader.ts [hunk 9]
```diff
@@ -0,0 +1,5 @@
+import { AwilixContainer } from 'awilix'
```
Comment (olivermrbl) [inline src/loaders/loader.ts]: todo: Add in README in `src/loaders/README.md`

File: src/services/my-custom.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import { Lifetime } from "awilix"
```
Comment (olivermrbl) [inline src/services/my-custom.ts]: todo: Add in README in `src/services/README.md`

File: src/subscribers/my-subscriber.ts [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import MyCustomService from "../services/my-custom";
```
Comment (olivermrbl) [inline src/subscribers/my-subscriber.ts]: todo: Add in README in `src/subscribers/README.md`

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: src/models/custom.ts (hunk 7 above)
Comment (adrien2p) [reply src/models/custom.ts]: there is already an example in readne, you want me to replace it or add a second example?

Comment (olivermrbl) [reply src/models/custom.ts]: Replace if this is more up to date

Comment (adrien2p) [reply src/models/custom.ts]: done it 👍

**Review by olivermrbl — APPROVED** [review]
This is great - nice work!

**Pull request merged**